src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java

```diff
@@ -228,19 +230,20 @@ public static void createDatasetIfNotExists(BigQuery bigQuery, DatasetId dataset
    * @param location     Location for this bucket.
    * @param cmekKeyName  CMEK key to use for this bucket.
    * @param errorMessage Supplier for the error message to output if the bucket could not be created.
-   * @throws IOException if the bucket could not be created.
    */
   private static void createBucket(Storage storage, String bucket, @Nullable String location,
                                    @Nullable CryptoKeyName cmekKeyName,
-                                   Supplier<String> errorMessage) throws IOException {
+                                   Supplier<String> errorMessage) {
     try {
       GCPUtils.createBucket(storage, bucket, location, cmekKeyName);
     } catch (StorageException e) {
       if (e.getCode() != 409) {
         // A conflict means the bucket already exists
         // This most likely means multiple stages in the same pipeline are trying to create the same dataset.
         // Ignore this and move on, since all that matters is that the dataset exists.
-        throw new IOException(errorMessage.get(), e);
+        throw ErrorUtils.getProgramFailureException(
+          new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN), "Bucket already exists", errorMessage.get(),
+          ErrorType.UNKNOWN, true, e);
```

Comment:
we can decide on error type here based on status code: see https://github.com/data-integrations/google-cloud/blob/601f62bd0005255e75bff19db8711a8898358d49/src/main/java/io/cdap/plugin/gcp/common/GCPErrorDetailsProvider.java#L69

Reply:
updated !

```java
try {
  GCPUtils.createBucket(storage, bucket, location, cmekKeyName);
} catch (StorageException e) {
  if (e.getCode() != 409) {
    // A conflict means the bucket already exists
    // This most likely means multiple stages in the same pipeline are trying to create the same dataset.
    // Ignore this and move on, since all that matters is that the dataset exists.
    ErrorUtils.ActionErrorPair pair = ErrorUtils.getActionErrorByStatusCode(e.getCode());
    String errorReason = String.format("%s %s %s", e.getCode(), e.getMessage(), pair.getCorrectiveAction());
    throw ErrorUtils.getProgramFailureException(
      new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN), errorReason, errorMessage.get(),
      pair.getErrorType(), true, e);
  }
}
```